source/agora/common/TransactionPool.d

```diff
@@ -301,6 +301,68 @@ public class TransactionPool
 
         return true;
     }
+
+    /***************************************************************************
+
+        Check if a transaction hash exists in the transaction pool.
+
+        Params:
+            tx = the transaction hash
+
+        Returns:
+            true if the has transaction hash.
+
+    ***************************************************************************/
+
+    public bool hasTransactionHash (const ref Hash tx) @trusted
+    {
+        auto results = this.db.execute("SELECT key, val FROM tx_pool " ~
```

Comment:
Technically you do not need the value here, so you could do `SELECT key FROM tx_pool WHERE key = ?`

Reply:
right, Suggested change